qiskit/qpy/interface.py

```diff
@@ -284,6 +285,15 @@ def load(
             )
         )
 
+    config = user_config.get_config()
+    
+    min_qpy_version = config.get("min_qpy_version")
+    if min_qpy_version is not None and common.QPY_VERSION < int(min_qpy_version):
```

Comment:
This comparison isn't correct. `common.QPY_VERSION` is a constant to indicate the max supported QPY version in Qiskit. You need to be comparing against the version in the payload header. You have it correct in the error message.